hyperon_das_atomdb/database.py

```diff
@@ -99,7 +100,7 @@ def _add_link(self, link_params: Dict[str, Any], toplevel: bool = True) -> Dict[
         ]
 
         if any(
-            item in reserved_parameters or item.startswith('key') for item in link_params.keys()
+            item in reserved_parameters or re.match(r'key_\d+', item) for item in link_params.keys()
```

Comment:
Regular expressions are expensive, specially if you don't compile them first. We can let them here and just add another issue to fix this later (I'll add a comment in this card we already have: https://github.com/singnet/das-atom-db/issues/57) but before merging you should compile this expression once e.g. before declaring this class in this file or in some imported file, as you prefer. You need to do something like

```
regexp = re.compile(<your expression>)
```

And then use it here like this `if regexp.search()`

I wrote this example without looking at the docs so please confirm this is the best method to call before using it.